### ISSUE TITLE
space:pairs() must pass key and opts parameters to primary index :pairs()

### ISSUE THREAD

**Issue by Gerold103** [tarantool/doc#71]

**Comment by pgulutzan** [comment]
I think this means: document space-object:pairs([key-value [,{option(s)]]}).
That's valid, since (although it's optional not mandatory) a key and an iterator option can be passed.
But please explain:
box.schema.space.create('t')
box.space.t:create_index('i')
box.space.t:insert{1}, box.space.t:insert{2}, box.space.t:insert{3}
box.space.t:select(1,{iterator='GT',limit=1})
for k, v in box.space.t:pairs(1,{iterator='GT',limit=1}) do print(v[1]) end
Notice that the select() request returns one tuple, but the pairs() request returns two tuples.
In other words, it appears that the "limit" option has been ignored.
So: we shouldn't say "opts" (plural), we should say "opt" or just "iterator-type",
as we do for indexes
https://tarantool.org/doc/book/box/box_index.html#lua-function.index_object.pairs
or am I misunderstanding what you mean by "opts"?

**Comment by rtsisyk** [comment]
>I think this means: document space-object:pairs([key-value [,{option(s)]]}).
>That's valid, since (although it's optional not mandatory) a key and an iterator option can be passed.
>But please explain:
>box.schema.space.create('t')
>box.space.t:create_index('i')
>box.space.t:insert{1}, box.space.t:insert{2}, box.space.t:insert{3}
>box.space.t:select(1,{iterator='GT',limit=1})
>for k, v in box.space.t:pairs(1,{iterator='GT',limit=1}) do print(v[1]) end
>Notice that the select() request returns one tuple, but the pairs() request returns two tuples.
>In other words, it appears that the "limit" option has been ignored.

`limit` and `offset` are not supported by pairs(), only by select().

You can use functional syntax (luafun) to filter the iterator output:

```
tarantool> box.space.test:pairs(1, {iterator = 'GE' }):totable()
---
- - [1]
  - [2]
  - [3]
  - [4]
  - [5]
  - [6]
  - [7]
  - [8]
  - [9]
  - [10]
...

tarantool> box.space.test:pairs(1, {iterator = 'GE' }):drop(4):take(5):totable() -- filtered
---
- - [5]
  - [6]
  - [7]
  - [8]
  - [9]
...
```

---------------------------------------------------------------------------------------------------------

>So: we shouldn't say "opts" (plural), we should say "opt" or just "iterator-type",
>as we do for indexes
>https://tarantool.org/doc/book/box/box_index.html#lua-function.index_object.pairs
>or am I misunderstanding what you mean by "opts"?
>

Some Lua functions accept optional arguments table:

* box.schema.space.create(name, options)
* box.schema.role.create(name, opts)
* box.schema.user.create(name, opts)
* index:select(index, key, opts)
* index:pairs(key, opts)
...

Sometimes "opts" is used as a shortened version of "options".

I can update Lua code to use word "options" everywhere if you want.

---------------------------------------------------------------------------------------------------------

https://tarantool.org/doc/book/box/box_space.html#lua-function.space_object.select

> space_object:select(key)

space_object:select(key, options),

where options can be

* iterator = iterator-type (string or unsinged)
* offset = offset (unsigned)
* limit = limit (unsigned)

---------------------------------------------------------------------------------------------------------

https://tarantool.org/doc/book/box/box_index.html#lua-function.index_object.pairs

> index_object:pairs(bitset-value | search-value, iterator-type)

index_object:pairs(bitset-value | search-value, iterator-type | options)

where options can be:

 * iterator = iterator-type (string or unsinged)

BTW, it seems that index_object:pairs(key, iterator-type) is actually broken:

```
tarantool> box.space.test:pairs(5, 'GE'):totable()
---
- - [5]
...

tarantool> box.space.test:pairs(5, {iterator = 'GE' }):totable()
---
- - [5]
  - [6]
  - [7]
  - [8]
  - [9]
  - [10]
...
```

I opened a [ticket](https://github.com/tarantool/tarantool/issues/1875) for this problem [1].
Please document " iterator-type | options" syntax, we will fix this bug soon.

---------------------------------------------------------------------------------------------------------

I prefer "index" and "space" to "index_object" and "space_object" in the documentation.

**Comment by pgulutzan** [comment]
Thank you, but this only confirms what I said:
we shouldn't say "opts" (plural), we should say "opt" or just "iterator-type",
as we do for indexes
https://tarantool.org/doc/book/box/box_index.html#lua-function.index_object.pairs
... which is strictly about the manual. I am definitely not requesting that anyone should change Lua code.
